component-runtime-impl/src/main/java/org/talend/sdk/component/runtime/record/RecordImpl.java

```diff
@@ -115,25 +115,37 @@ public BuilderImpl(final Schema providedSchema) {
             this.providedSchema = providedSchema;
         }
 
-        private void validateTypeAgainstProvidedSchema(final String name, final Schema.Type type, final Object value) {
-            if (providedSchema == null) {
-                return;
-            }
-            if (entryIndex == null) {
-                entryIndex = providedSchema.getEntries().stream().collect(toMap(Schema.Entry::getName, identity()));
+        private Schema.Entry findMandatoryEntry(final String name) {
+            assert this.providedSchema != null;
+            if (this.entryIndex == null) {
+                this.entryIndex = providedSchema.getEntries().stream().collect(toMap(Schema.Entry::getName, identity()));
             }
-            final Schema.Entry entry = entryIndex.get(name);
+            final Schema.Entry entry = this.entryIndex.get(name);
             if (entry == null) {
                 throw new IllegalArgumentException(
                         "No entry '" + name + "' expected in provided schema: " + entryIndex.keySet());
             }
-            if (entry.getType() != type) {
-                throw new IllegalArgumentException(
-                        "Entry '" + name + "' expected to be a " + entry.getType() + ", got a " + type);
+            return entry;
+        }
+
+        private Schema.Entry findOrBuildEntry(final String name, final Schema.Type type, boolean nullable) {
+            if (providedSchema == null) {
+                return new SchemaImpl.EntryImpl.BuilderImpl().withName(name).withType(type).withNullable(nullable).build();
+            }
+            return this.findMandatoryEntry(name);
+        }
+
+        private Schema.Entry validateTypeAgainstProvidedSchema(final String name, final Schema.Type type, final Object value) {
```

Comment:
Here, 
- the function name do not match the goal of the function and provided type is useless.
- the function has a side effect by creating a new entry and it doesn't match initial goal.

Reply:
- type checking added (don't know if it's usefull, assertType do also this control)
- this method never create entry.

Reply:
you're right on second point ;-)